### ISSUE TITLE
Keyboard-Only Focus Styles

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#6879]
With accessibility coming to spartacus, all focusable elements are styled with an outline. While this might be best practices for accessibility (although I'd favour dedicated styles which could be equally noticed), it is surely not the experience that "mouse-users" would expect. 

The strong focus highlights would be OK if the mouse-user is partially using the keyboard (i.e. navigating through a list). But when the mouse is used to navigate through the experience, having such a dominant focus outline would be considered awkward. 

The industry agrees to that starting point and a keyboard-only focus style feature was proposed, using the `:focus-visible` pseudo selector, next to the `:focus` selector. The intent is that you can style elements when the focus was driven by the keyboard rather than the mouse. 

This is the right direction, but so far [only firefox implemented this](https://caniuse.com/#search=focus-visible). 

With this ticket we'll evaluate a "polyfill" implementation for the focus-visible.

**Comment by tobi-or-not-tobi** [comment]
QA
- enable calydon theme (if not already standardized)
- use the keyboard to navigate the storefront, and confirm that the UI is using the focusable blue outline
- use the mouse to navigate the storefront, and confirm that the UI  is not showing the focusable outline
- alternate both navigation options (keyboard and mouse) and confirm that the outline will be shown as soon as the mouse is used. 
- test this generic behaviour randomly throughout the storefront; header, footer, cart, checkout, PLP, PDP.

**Comment by KateChuen** [comment]
Tested on dev10. Noticed these, @tobi-or-not-tobi please confirm if they are known, intended or require new tickets:

**Mouse navigation**
On the add-to-cart modal, the outline is showing for mouse navigation. If I add an item to cart, we will momentarily see the outline on the "Close" button.
<img width="788" alt="Screen Shot 2020-03-23 at 11 46 44 AM" src="https://user-images.githubusercontent.com/45409515/77335627-c885ed00-6cfc-11ea-9576-82867bc747bf.png">

Then if we use the mouse to click on the - + buttons to change the quantity of the item, we will see the outline there too.
<img width="776" alt="Screen Shot 2020-03-23 at 11 47 09 AM" src="https://user-images.githubusercontent.com/45409515/77335566-b73ce080-6cfc-11ea-98ac-5b20cf01f1c7.png">

Same thing on the "Remove" button
<img width="773" alt="Screen Shot 2020-03-23 at 1 40 36 PM" src="https://user-images.githubusercontent.com/45409515/77346056-f888bc80-6d0b-11ea-818e-c09ee0500542.png">

**Keyboard navigation**
Using the keyboard navigation, there's no outline on the - + buttons on the cart page and add-to-cart modal. So I cannot change the qty of the item, unless I change it in the text box. 
<img width="827" alt="Screen Shot 2020-03-23 at 1 29 36 PM" src="https://user-images.githubusercontent.com/45409515/77346198-41407580-6d0c-11ea-96cc-b4f5e4a47879.png">

**Comment by tobi-or-not-tobi** [comment]
The visible focus and autofocus on the cart item modal are caused by the ngBootstrap component. We're on our way to replace this component, which is why I don't like to implement work arounds for this. 

The lack of visual focus on the increment and decrement buttons is not added intentionally, as the idea is that the keyboard user can use arrow-up / arrow-down, or enter numeric values directly. @developpeurweb I believe you've been involved here, but otherwise please raise your voice.

**Comment by developpeurweb** [comment]
@tobi-or-not-tobi 
Sorry for the delay on answering this. On a strict adherence to Accessibility Guidelines: This OK, because:

1. Provides 2 accessible alternatives for keyboard users.
2. Focus goes directly to the input field to do the change manually without passing through the buttons.

But, on a personal note, and with client customization in mind, by browsing high-end retailers having this type of item-counter, I discovered different uses, all of them accessible, but with different approaches and customizations.

1. If clients decide to use **ONLY** the buttons (and not manual input) to change quantity, then our item-counter will have to be improved as our buttons are not currently focusable.
2. If clients decide make keyboard-accessible **ALL 3** ways of changing quantity, then our item-counter will have to be improved as our buttons are not currently focusable.

Therefore, maybe we can **improve this component in future versions** to support ALL client customizations.

More details on examples here: https://cxwiki.sap.com/display/spar/Item-counter+Accessibility

**Comment by developpeurweb** [comment]
Ticket for Screen Reader future improvements on `item-counter` component GH-7122
